### PR TITLE
feat: support SRV record

### PR DESCRIPTION
**Pull request by spacewander**
Signed-off-by: spacewander <spacewanderlzx@gmail.com>

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [x] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

File: apisix/core/dns/client.lua [hunk 1]
```diff
@@ -29,6 +31,67 @@ local _M = {
...
+            local copy = table.deepcopy(res)
+            copy.weight = weight / count
+            copy.port = answer.port
+            insert_tab(resolved_answers, copy)
+        end
+    end
+
+    -- find the least common multiple of the counts
+    local lcm = answer_to_count[answers[1]]
+    for i = 1, #answers do
```
Comment (tokers) [inline apisix/core/dns/client.lua:82]: We can iterate from `i = 2`.

File: apisix/core/dns/client.lua [hunk 2]
```diff
@@ -29,6 +31,67 @@ local _M = {
...
+        local resolved, err = client.resolve(answer.target)
+        if not resolved then
+            local msg = "failed to resolve SRV record " .. answer.target .. ": " .. err
+            return nil, msg
+        end
+
+        log.info("dns resolve SRV ", answer.target, ", result: ",
+                 json.delay_encode(resolved))
+
+        local weight = answer.weight
```
[the remark this answers is not in the document]
Comment (spacewander) [reply apisix/core/dns/client.lua:64]: Added: https://github.com/apache/apisix/pull/3686/files#diff-eeebb3321eff56ec1f7b883a6a08b29bbc6359cdd06407eca329772b9a619676R122

File: apisix/core/dns/client.lua (hunk 1 above)
Comment (spacewander) [reply apisix/core/dns/client.lua:82]: Solved

**Review by membphis — APPROVED** [review]

File: docs/en/latest/dns.md [hunk 3]
```diff
@@ -74,3 +74,60 @@ If you want to specify the port for the upstream server, you can add it to the `
...
+}
+```
+
+is the same as:
+
+```json
+{
+    "id": 1,
+    "type": "roundrobin",
+    "nodes": {
```
Comment (membphis) [inline docs/en/latest/dns.md:112]: should use `[` here, it an array object here.

**Review by membphis — CHANGES_REQUESTED** [review]

File: docs/en/latest/dns.md (hunk 3 above)
Comment (spacewander) [reply docs/en/latest/dns.md]: @membphis 
Solved.

**Review by membphis — APPROVED** [review]

**Review by tokers — APPROVED** [review]

**Pull request merged**